rs/registry/canister/src/mutations/node_management/do_replace_operator.rs

```diff
@@ -0,0 +1,628 @@
+                return Err(format!(
+                    "{}do_replace_operator: Node {} does not belong to node operator {}",
+                    LOG_PREFIX, node_id, old_operator_id
+                ));
+            }
+
+            node_records.insert(*node_id, node_record);
+        }
+
+        Ok(node_records)
+    }
+
+    /// Fetches all the node operator records for a single
+    /// node provider.
+    fn maybe_fetch_operators_for_provider(
+        &self,
+        provider_id: &PrincipalId,
+    ) -> Result<Vec<NodeOperatorRecord>, String> {
+        let operators: Vec<_> = self
+            .get_node_operators_and_dcs_of_node_provider(*provider_id)
+            .map(|operators_and_dcs| operators_and_dcs.into_iter().map(|(_, o)| o).collect())
+            .map_err(|e| format!("{}do_replace_operator: {:?}", LOG_PREFIX, e))?;
+
+        if operators.is_empty() {
+            return Err(format!(
+                "{}do_replace_operator: Unknown node provider {}",
+                LOG_PREFIX, provider_id
+            ));
+        }
+
+        Ok(operators)
+    }
+
+    /// Tries to find node operator records for `old_operator_id` and `new_operator_id` that
+    /// have to be within `operators`.
+    ///
+    /// Function will error out if the node operator records are not within the same data
+    /// center which is requred for replacing node operator functionality.
+    fn maybe_find_operator_records(
+        &self,
+        operators: Vec<NodeOperatorRecord>,
+        provider: &PrincipalId,
+        old_operator_id: &PrincipalId,
+        new_operator_id: &PrincipalId,
+    ) -> Result<(NodeOperatorRecord, NodeOperatorRecord), String> {
+        match (
+            find_node_operator_record_for_provider(&operators, old_operator_id, provider),
+            find_node_operator_record_for_provider(&operators, new_operator_id, provider),
+        ) {
+            (Ok(old_operator_record), Ok(new_operator_record))
+                if old_operator_record.dc_id != new_operator_record.dc_id =>
+            {
+                Err(format!("{}do_replace_operator: Old node operator and new node operator are in different data centers. Old node operator {} is in {} but the new node operator {} is in {}", LOG_PREFIX, old_operator_id, old_operator_record.dc_id, new_operator_id, new_operator_record.dc_id))
+            }
+            (Ok(old_operator_record), Ok(new_operator_record)) => {
+                Ok((old_operator_record.clone(), new_operator_record.clone()))
+            }
+            (Err(e), _) => Err(e),
+            (_, Err(e)) => Err(e),
+        }
+    }
+}
+
+/// Helper function which tries to find find a single node operator record
+/// within an array of `operators`, returning either a reference to the found
+/// record or an error.
+fn find_node_operator_record_for_provider<'a>(
+    operators: &'a [NodeOperatorRecord],
+    operator_id: &PrincipalId,
+    provider: &PrincipalId,
```

Comment:
It is a little confusing when an argument isn't "really" used (in this case, only to make error string more helpful). Therefore, a comment is probably warranted.